utilities/transactions/write_prepared_txn_db.cc

```diff
@@ -273,30 +270,34 @@ bool WritePreparedTxnDB::IsInSnapshot(uint64_t prep_seq,
       found = std::binary_search(vec.begin(), vec.end(), prep_seq);
     }
     if (!found) {
-      ROCKSDB_LOG_DETAILS(
+      ROCKS_LOG_DETAILS(
           info_log_, "IsInSnapshot %" PRIu64 " in %" PRIu64 " returns %" PRId32,
           prep_seq, snapshot_seq, 1);
       return true;
     }
   }
   // (ii) it the case: it is committed but after the snapshot_seq
-  ROCKSDB_LOG_DETAILS(
-      info_log_, "IsInSnapshot %" PRIu64 " in %" PRIu64 " returns %" PRId32,
-      prep_seq, snapshot_seq, 0);
+  ROCKS_LOG_DETAILS(info_log_,
+                    "IsInSnapshot %" PRIu64 " in %" PRIu64 " returns %" PRId32,
+                    prep_seq, snapshot_seq, 0);
   return false;
 }
 
 void WritePreparedTxnDB::AddPrepared(uint64_t seq) {
-  ROCKSDB_LOG_DETAILS(info_log_, "Txn %" PRIu64 " Prepareing", seq);
-  // TODO(myabandeh): Add a runtime check to ensure the following assert.
+  ROCKS_LOG_DETAILS(info_log_, "Txn %" PRIu64 " Prepareing", seq);
   assert(seq > max_evicted_seq_);
+  if (seq <= max_evicted_seq_) {
+    throw std::runtime_error(
```

Comment:
Can we return status instead of crashing?

Reply:
If we do that we should be able to recover from error as well. If the write succeeds but we cannot update the prepare heap, there is no existing mechanism to recover from that error.
Moreover Status seems more suitable for rare, but feasible error cases. This case is almost impossible to occur and is added there mostly for theoretical correctness.